js/ai/src/tool.ts

```diff
@@ -55,8 +61,37 @@ export type ToolAction<
     replyData: z.infer<O>,
     options?: { metadata?: Record<string, any> }
   ): ToolResponsePart;
+  /**
+   * restart constructs a tool request corresponding to the provided interrupt tool request
+   * that will then re-trigger the tool after e.g. a user confirms. The `resumedMetadata`
+   * supplied to this method will be passed to the tool to allow for custom handling of
+   * restart logic.
+   *
+   * @param interrupt The interrupt tool request you want to restart.
+   * @param resumedMetadata The metadata you want to provide to the tool to aide in reprocessing. Defaults to `true` if none is supplied.
+   * @param options Additional options for restarting the tool.
+   */
+  restart(
+    interrupt: ToolRequestPart,
+    resumedMetadata?: any,
```

Comment:
restart and resume are used interchangeably... should probably be consistent to avoid confusion. Maybe call it `restartMetadata`? and `restarted`?

Reply:
I made the tool API `(input, {interrupt, resumed})` because `(input, {interrupt, restarted})` doesn't really feel as nice...but maybe it's fine?

Another option could be going back to the original `confirm` and make the argument `confirmed`. (Then we could have resume + complete + confirm?)

I think restart may still be the most accurate tho...